src/Nethermind/Nethermind.JsonRpc/Modules/BlockFinderExtensions.cs

```diff
@@ -76,9 +76,23 @@ public static SearchResult<Block> SearchForBlock(this IBlockFinder blockFinder,
                 block = blockFinder.FindBlock(blockParameter);
             }
 
-            return block == null && !allowNulls
-                ? new SearchResult<Block>($"Block {blockParameter.BlockHash?.ToString() ?? blockParameter.BlockNumber?.ToString() ?? blockParameter.Type.ToString()} could not be found", ErrorCodes.ResourceNotFound)
-                : new SearchResult<Block>(block);
+            if (block != null)
+            {
+                if (!block.IsPostMerge && (blockParameter.Equals(BlockParameter.Finalized) ||
```

Comment:
how to determine if it's pre-merge? is this okay?

Reply:
Finalized is zero hash before the merge

Reply:
https://github.com/ethereum/execution-apis/pull/200/files#:~:text=Before%20the%20merge,number%20or%20tag%22%2C